### ISSUE TITLE
Redundancy of optimiser.zero_grad()?

### ISSUE THREAD

**Issue by meinternational** [intel-isl/MultiObjectiveOptimization#34]
There are two calls of optimizer.zero_grad()

https://github.com/intel-isl/MultiObjectiveOptimization/blob/5d8e8343c56cf3184081f71cc4a661af64e7cf7e/multi_task/train_multi_task.py#L114

https://github.com/intel-isl/MultiObjectiveOptimization/blob/5d8e8343c56cf3184081f71cc4a661af64e7cf7e/multi_task/train_multi_task.py#L146

before optimizer.zero_grad() is called finally in line: 

https://github.com/intel-isl/MultiObjectiveOptimization/blob/5d8e8343c56cf3184081f71cc4a661af64e7cf7e/multi_task/train_multi_task.py#L173

and optimizer.step() is applied.

Are the first two calls of optimizer.zero_grad() necessary or redundant? 

Thx

**Comment by ozansener** [comment]
In the current implementation, both are necessary. You can see that the loss is modified in between those two. 

However, there is a way to implement it with using only one zero_grad by modifying the gradients directly. If the computation is an issue, you can compute the gradients, then compute the weights and finally replace gradients with weighted gradients and call the step.

**Comment by meinternational** [comment]
Thank you very much for your swift response.
